File: app/src/main/java/com/example/customstarrating/MainActivity.java
package com.example.customstarrating;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ImageView addButton = findViewById(R.id.button_plus);
        ImageView minusButton = findViewById(R.id.button_minus);
        final RatingsLayout ratingsLayout = findViewById(R.id.star_rating_layout);

        addButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ratingsLayout.addStar();
            }
        });

        minusButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ratingsLayout.removeStar();
            }
        });
    }
}

File: app/src/main/java/com/example/customratingbar/ui/MainActivity.kt
package com.example.customratingbar.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import com.example.customratingbar.R
import com.example.customratingbar.views.RatingBarView
import com.example.customratingbar.fragments.RatingFragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_add_rating.setOnClickListener {
            val ratingFragment:DialogFragment = RatingFragment()
             ratingFragment.show(supportFragmentManager, "Rating Fragment")

        }
    }
    fun addView(){
        var textView= TextView(this)
        textView.text="${RatingBarView.name} ${RatingBarView.rating}"
        textView.textSize=20f
        layout_ratings.addView(textView)

    }

}
